infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -342,72 +367,104 @@ private void handleTokenQuery(CorfuPayloadMsg<TokenRequest> msg,
             token = new Token(sequencerEpoch, globalLogTail - 1);
             streamTails = new HashMap<>(streams.size());
             for (UUID stream : streams) {
-                streamTails.put(stream, streamTailToGlobalTailMap.getOrDefault(stream, Address.NON_EXIST));
+                streamTails.put(stream,
+                        streamTailToGlobalTailMap.getOrDefault(stream, Address.NON_EXIST));
             }
         }
 
-        r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(new TokenResponse(
-                TokenType.NORMAL, TokenResponse.NO_CONFLICT_KEY,
-                TokenResponse.NO_CONFLICT_STREAM, token, Collections.emptyMap(), streamTails)));
-
+        // Note: we reuse the request header as the ignore_cluster_id and
+        // ignore_epoch fields are the same in both cases.
+        ResponseMsg response = getResponseMsg(
+                req.getHeader(),
+                getTokenResponseMsg(TokenType.NORMAL,
+                        TokenResponse.NO_CONFLICT_KEY,
+                        TokenResponse.NO_CONFLICT_STREAM,
+                        token,
+                        Collections.emptyMap(),
+                        streamTails)
+        );
+
+        r.sendResponse(response, ctx);
     }
 
-    @ServerHandler(type = CorfuMsgType.SEQUENCER_TRIM_REQ)
-    public void trimCache(CorfuPayloadMsg<Long> msg, ChannelHandlerContext ctx, IServerRouter r) {
+    @RequestHandler(type = PayloadCase.SEQUENCER_TRIM_REQUEST)
+    public void trimCache(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
         log.info("trimCache: Starting cache eviction");
```

Comment:
Add @Nonnull annotation for the request handler methods could improve code inspection for inferring nullity﻿. Please check this comment
https://github.com/CorfuDB/CorfuDB/pull/2827#discussion_r535539159